apps/src/lib/tools/jsinterpreter/CustomMarshalingInterpreter.js

```diff
@@ -3,29 +3,17 @@ const codegen = require('../../../codegen');
 const CustomMarshaler = require('./CustomMarshaler');
 
 module.exports = class CustomMarshalingInterpreter extends Interpreter {
-
   constructor(code, customMarshaler, opt_initFunc) {
-    super(code);
-    if (!(customMarshaler instanceof CustomMarshaler)) {
-      throw new Error("You must provide a CustomMarshaler to CustomMarshalingInterpreter");
-    }
-    this.customMarshaler = customMarshaler;
-
-    // Because of the way the interpreter is written, we have to re-do
-    // this work from the parent class's constructor because it needs
-    // customMarshaler to have been set before-hand. Unfortunately, javascript
-    // does not allow us to set customMarshaler until after the parent constructor
-    // has finished.
-    // TODO (pcardune): change the initialization process of the Interpreter
-    // class to make this work better.
-    this.initFunc_ = opt_initFunc;
-    this.globalScope = this.createScope(this.ast, null);
-    this.stateStack = [{
-      node: this.ast,
-      scope: this.globalScope,
-      thisExpression: this.globalScope,
-      done: false
-    }];
+    super(code, (thisInterpreter, scope) => {
+      if (!(customMarshaler instanceof CustomMarshaler)) {
+        throw new Error("You must provide a CustomMarshaler to CustomMarshalingInterpreter");
+      }
+      thisInterpreter.customMarshaler = customMarshaler;
+      thisInterpreter.globalScope = scope;
```

Comment:
Not really an issue, but is it just me or is it weird that the interpreter instance is passed to the callback?  Do I understand correctly that `this === thisInterpreter` here?  If so, I wonder if this would read a little cleaner as
```jsx
super(code, (_, scope) => {
  // ...
  this.customMarshaler = customMarshaler;
  this.globalScore = scope;
```
just to avoid this question in the future?